dali/operators/reader/loader/webdataset_loader.h

```diff
@@ -32,8 +31,37 @@ namespace dali {
 namespace detail {
 namespace wds {
 
+/**
+ * The index version string should look like:
+ * v<one-or-two-digits>.<one-or-two-digits>
+ * e.g. v1.2, v10.1, v0.22, v12.12
+ */
+inline int DLL_PUBLIC CalcIndexVersion(const std::string& index_version_string) {
+  using std::isdigit;
+  auto it = index_version_string.rbegin();
+  int major = 0, minor = 0;
+  if (!isdigit(*it)) return -1;
+  minor += *it++ - '0';
+  if (isdigit(*it)) {
+    minor += (*it++ - '0') * 10;
+  }
+  if (*it++ != '.') return -1;
+  if (!isdigit(*it)) return -1;
+  major += *it++ - '0';
+  if (isdigit(*it)) {
+    major += (*it++ - '0') * 10;
+  }
+  if (*it++ != 'v') return -1;
+  if (it != index_version_string.rend()) return -1;
+  return major * 1000 + minor;
+}
+
+// Minor part of the index version may be at most two-digit.
 const std::string kCurrentIndexVersion = "v1.2";  // NOLINT
-const std::unordered_set<std::string> kSupportedIndexVersions = {"v1.1", kCurrentIndexVersion};
+const std::set<int> kSupportedIndexVersions = {
+    1002,  // v1.2
+    1001,  // v1.1
```

Comment:
I think we've discussed something like `MakeIndexVersion(1, 2)`. Any particular reason it went out of favor?

Reply:
Added.

Reply:
Not seeing anything.